server/controllers/conversation.js

```diff
@@ -0,0 +1,142 @@
+const mongoose = require('mongoose');
+const asyncHandler = require('express-async-handler');
+const Conversation = require('../models/Conversation');
+const Message = require('../models/Message');
+const User = require('../models/User');
+
+exports.createConversation = asyncHandler(async (req, res, next) => {
+  const { from, to } = req.body;
+
+  try {
+    if (!from || !to) {
+      res.status(400).json({
+        error: 'from and to are required',
+      });
+      throw new Error('from and to are required');
+    }
+
+    const fromUser = await User.findOne({ email: from });
+    const toUser = await User.findOne({ email: to });
+    if (!fromUser || !toUser) {
+      res.status(400).json({
+        error: 'unknown user',
+      });
+      throw new Error('unknown user');
+    }
+
+    const conversation = await Conversation.create({
+      fromUser: fromUser.get('_id'),
+      toUser: toUser.get('_id'),
+    });
+
+    res.status(201).json({
+      success: true,
+      conversation,
+    });
+  } catch (error) {}
```

Comment:
Maybe log server errors to the console or send it as a response?1

Ditto throughout

Reply:
Done